agent/src/com/cloud/agent/AgentShell.java

```diff
@@ -107,7 +108,16 @@ public String getPod() {
 
     @Override
     public String getHost() {
-        return _host;
+        String[] hosts = _host.split(",");
+        if (_hostCounter >= hosts.length) {
+            _hostCounter = 0;
+        }
+        s_logger.info("Connecting to host: " + hosts[_hostCounter % hosts.length]);
+        return hosts[_hostCounter++ % hosts.length];
+    }
+
+    public void setHost(final String host) {
```

Comment:
For historic reasons, I simply created getters/setters for `_host`, this method is only used for writing a unit test.